### PR TITLE
Fixing incorrect (recent?) refactoring: reverting devicePriorities to be vector and respect the order, as opposed to the unordered_map that effectively ignores the priorities

### PR DESCRIPTION
**Pull request by myshevts**
@apankratovantonp , when you have replaced the device priorities to be unordered_map and not the vector, the priorities were essentially ignored 🥇 . I strongly suggest you do a code review with me before doing refactoring like this.

### CODE REVIEW

**Comment by ilya-lavrenov** [comment]
@myshevts please, submit the same changes to release 2021.1

**Comment by ilya-lavrenov** [comment]
> I strongly suggest you do a code review with me before doing refactoring like this.

code review is done. All PRs have +1 from you (see gitlab).
In order to avoid such situations, could you please add a test for this behavior?

File: inference-engine/src/multi_device/multi_device.cpp [hunk 1]
```diff
@@ -258,8 +259,8 @@ void MultiDeviceExecutableNetwork::SetConfig(const std::map<std::string, Inferen
         assert(multiPlugin != nullptr);
         auto metaDevices = multiPlugin->ParseMetaDevices(priorities->second, {});
 
-        if (std::any_of(metaDevices.begin(), metaDevices.end(), [](const std::pair<DeviceName, DeviceInformation> & kvp) {
```
Comment (apankratovantonp) [inline inference-engine/src/multi_device/multi_device.cpp:261]: kvp - key value pair. Use other name now

Comment (myshevts) [reply inference-engine/src/multi_device/multi_device.cpp:261]: this is minor, and would require changing/re-testing similar fix to the release branch (https://github.com/openvinotoolkit/openvino/pull/2251) which might miss the code-freeze then

**Comment by myshevts** [comment]
> > I strongly suggest you do a code review with me before doing refactoring like this.
> 
> code review is done. All PRs have +1 from you (see gitlab).
> In order to avoid such situations, could you please add a test for this behavior?

good point: https://jira.devtools.intel.com/browse/CVS-38404. The only complexity that we will need mocking 2 or more devices to simulate the things

**Comment by myshevts** [comment]
Rebuild

**Comment by myshevts** [comment]
rebuild

**Review by ilya-lavrenov — APPROVED** [review]

**Comment by ilya-lavrenov** [comment]
CI is red

**Comment by myshevts** [comment]
> CI is red

this is Myriad sticks that are down, will need to "rebuild" again

**Comment by myshevts** [comment]
rebuild

**Comment by myshevts** [comment]
rebuild

**Pull request merged**